### PR TITLE
Feature/159

### LINKED ISSUE #159: Images should be hashed, and saved in the ImageHash row
**Describe the solution you'd like**
Images should be hashed, and stored in the database table.
As this should be used when implementing hashing, as it can be used to compare images.

**Additional context**
This is related to #47 but this neeeds to be implemented instead.

### PR DESCRIPTION
**Pull request by ggAlexS**
Added md5 hashing for images and hashes are now saved in the database Closes #159

### CODE REVIEW

**Comment by Determinatoro** [comment]
Please use the following checklist, @Huften. Feel free to check it off as you go along.

**Code Design** 

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [ ] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, run the web-api locally as described in https://aau-giraf.github.io/wiki/development/rest_api_development/BuildAndRunLocally/ 
- [ ] The web-api runs without issues
- [ ] The functionality that the code claims to implement, are actually implemented
- [ ] The integration tests runs successfully. Refer to https://github.com/aau-giraf/web-api/tree/master/GirafIntegrationTest for how to run the tests

**Comment by Determinatoro** [comment]
Please use the following checklist, @smh87. Feel free to check it off as you go along.

**Code Design** 

- [ ] The code is in the right place? (Both in terms of folder structure and class structure)
- [ ] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [ ] This code could not, to the best of my knowledge, have reused existing code
- [ ] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [ ] Names are meaningful and self-documenting
- [ ] It is understandable, by reading the code, what it does
- [ ] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [ ] Has all the added code been properly commented

Only check one of the items below:

- [ ] The functionality does **not** need to be covered by tests
- [ ] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, run the web-api locally as described in https://aau-giraf.github.io/wiki/development/rest_api_development/BuildAndRunLocally/ 
- [ ] The web-api runs without issues
- [ ] The functionality that the code claims to implement, are actually implemented
- [ ] The integration tests runs successfully. Refer to https://github.com/aau-giraf/web-api/tree/master/GirafIntegrationTest for how to run the tests

**Comment by smh87** [comment]
Please seee this befor merging this issue https://stackoverflow.com/questions/3748/storing-images-in-db-yea-or-nay
It is not recommonded to store a hole image in the database. But stor a path to a picture.

File: GirafRest/Controllers/PictogramController.cs [hunk 1]
```diff
@@ -346,13 +346,12 @@ public async Task<ActionResult> SetPictogramImage(long id)
 
 
 
-                pictogram.ImageHash = image.GetHashCode().ToString();
+                pictogram.ImageHash = _giraf.GetHash(image);
```
Comment (smh87) [inline GirafRest/Controllers/PictogramController.cs:349]: We change in pictogram controller. This should be tested if it works.

**Review by smh87 — CHANGES_REQUESTED** [review]
There is a lot of The variable 'XXXXXX' is declared but never used. E.g.

 Check warning on line 341 in GirafRest/Controllers/PictogramController.cs
GitHub Actions / Unit Test (ubuntu-18.04)
GirafRest/Controllers/PictogramController.cs#L341
The variable 'uaex' is declared but never used

can we do something about it?

The PictogramController.cs is changed we should test if it works.

**Comment by ggAlexS** [comment]
These warnings have not been created by our changes

File: GirafRest/Controllers/PictogramController.cs (hunk 1 above)
Comment (ggAlexS) [reply GirafRest/Controllers/PictogramController.cs:349]: Gethash() is tested, and pictogramcontroller should have tests from when it was created

**Comment by ggAlexS** [comment]
> There is a lot of The variable 'XXXXXX' is declared but never used. E.g.
> 
> 
> 
>  Check warning on line 341 in GirafRest/Controllers/PictogramController.cs
> 
> GitHub Actions / Unit Test (ubuntu-18.04)
> 
> GirafRest/Controllers/PictogramController.cs#L341
> 
> The variable 'uaex' is declared but never used
> 
> 
> 
> can we do something about it?
> 
> 
> 
> The PictogramController.cs is changed we should test if it works.
> 
> 
> 
> 

These warnings have not been created by our changes

**Comment by smh87** [comment]
The PictogramController.cs is changed we should test if it works.

**Comment by JeppeKLau** [comment]
> 
> 
> The PictogramController.cs is changed we should test if it works.
> About the warnings we will create a new issue.

The Pictogram controller is already covered by tests. And the new function `GetHash()` is covered by unittests.

**Review by smh87 — APPROVED** [review]
okay if all the functions are covered, then we are ready to go

**Review by Rosmarium — APPROVED** [review]

**Pull request merged**